src/app/infobox/infobox.component.spec.ts

```diff
@@ -1,65 +1,57 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { HttpModule, JsonpModule } from '@angular/http';
 import { InfoboxComponent } from './infobox.component';
-import {AppComponent} from "../app.component";
-import {NavbarComponent} from "../navbar/navbar.component";
-import {IndexComponent} from "../index/index.component";
-import {ResultsComponent} from "../results/results.component";
-import {NotFoundComponent} from "../not-found/not-found.component";
-import {AdvancedsearchComponent} from "../advancedsearch/advancedsearch.component";
-import {SearchBarComponent} from "../search-bar/search-bar.component";
-import {FooterNavbarComponent} from "../footer-navbar/footer-navbar.component";
-import {AboutComponent} from "../about/about.component";
-import {ContactComponent} from "../contact/contact.component";
-import {ModalComponent} from "ng2-bs3-modal/components/modal";
-import {RouterTestingModule} from "@angular/router/testing";
-import {BrowserModule} from "@angular/platform-browser";
-import {CommonModule} from "@angular/common";
-import {FormsModule} from "@angular/forms";
-import {HttpModule, JsonpModule} from "@angular/http";
-import {KnowledgeapiService} from "../services/knowledgeapi.service";
-import {reducer} from "../reducers/index";
-import {StoreModule} from "@ngrx/store";
-import {StoreDevtoolsModule} from "@ngrx/store-devtools";
+import { RouterTestingModule } from "@angular/router/testing";
+import { ThemeService } from '../services/theme.service';
+import { KnowledgeapiService } from "../services/knowledgeapi.service";
+import { reducer } from "../reducers/index";
+import { StoreModule } from "@ngrx/store";
+import {MockKnowledgeApi} from "../shared/mocks/knowledge.mock";
 
-describe('InfoboxComponent', () => {
+describe('Component: InfoboxComponent', () => {
   let component: InfoboxComponent;
   let fixture: ComponentFixture<InfoboxComponent>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       imports: [
         RouterTestingModule,
-        BrowserModule,
-        CommonModule,
-        FormsModule,
         HttpModule,
         JsonpModule,
         StoreModule.provideStore(reducer),
-        StoreDevtoolsModule.instrumentOnlyWithExtension(),
       ],
       declarations: [
-        AppComponent,
-        NavbarComponent,
-        IndexComponent,
-        ResultsComponent,
-        NotFoundComponent,
-        AdvancedsearchComponent,
-        SearchBarComponent,
-        FooterNavbarComponent,
-        AboutComponent,
-        ContactComponent,
-        ModalComponent,
-        InfoboxComponent ],
+        InfoboxComponent,
+      ],
       providers: [
-        KnowledgeapiService
+        { provide: KnowledgeapiService, useValue: MockKnowledgeApi},
```

Comment:
Keep proper spacing between curly braces and imported elements. Please follow styles.